### ISSUE TITLE
[CLI] sky down fail to remove a cluster that is already removed manually

### ISSUE THREAD

**Issue by Michaelvll** [skypilot-org/skypilot#1783]
To reproduce:
1. `sky launch -c test`
2. `sky stop test`
3. terminate the cluster manually in the console
4. `sky down test`

It shows the following error for AWS:
```
E 03-15 20:08:35 cloud_vm_ray_backend.py:3106] Failed to terminate test-autostop. If you want to ignore this error and remove the cluster from the status table, use `sky down --purge`.
E 03-15 20:08:35 cloud_vm_ray_backend.py:3106] **** STDOUT ****
E 03-15 20:08:35 cloud_vm_ray_backend.py:3106] 
E 03-15 20:08:35 cloud_vm_ray_backend.py:3106] **** STDERR ****
E 03-15 20:08:35 cloud_vm_ray_backend.py:3106] 
E 03-15 20:08:35 cloud_vm_ray_backend.py:3106] An error occurred (InvalidParameterCombination) when calling the TerminateInstances operation: No instances specified
```

**Comment by Michaelvll** [comment]
This issue should be fixed by #1702

**Comment by Michaelvll** [comment]
This issue has been resolved for GCP and AWS by #2199 and #2121. We still need to fix it for other clouds as well.
